wa/instruments/perf.py

```diff
@@ -134,5 +146,38 @@ def update_output(self, context):
                                 metric = '{}_{}'.format(label, match.group(3))
                                 context.add_metric(metric, count, classifiers=classifiers)
 
+    def _update_output_record(self, context):
+        outdir = os.path.join(context.output_directory, 'perf')
+        self.collector.get_trace(outdir)
+
+        self.logger.info('Processing perf report reports.')
+        for host_file in os.listdir(outdir):
+            label = host_file.split('.rpt')[0]
+            host_file_path = os.path.join(outdir, host_file)
+            context.add_artifact(label, host_file_path, 'raw')
+            with open(host_file_path) as fh:
+                for line in fh:
+                    words = line.split()
+                    if len(words) >= 1:
+                        if words[0] == '#' and len(words) == 6:
+                            if words[4] == 'event':
+                                event_type = words[5]
+                                event_type = event_type.strip("'")
+                        if words[0] != '#' and '%' in words[0] and len(words) == 5:
+                            metric = 'perf/{}/{}/{}'.format(event_type, words[1], words[2].strip("[]"))
```

Comment:
Consider using classifiers rather than packing everything into the metric name.

Reply:
I've added the classifiers but also left it packed in the metric name. It seems that some of the output processors, e.g. csv, default to classifier output off and that will result to a lot of meaningless/duplicate entries in the results file by default options. Which doesn't seem likely friendly behavior. 

I am very open to peoples thoughts on what is the right behavior here thou.

Reply:
I had a thought today that change the metric to <event>/<app>  which should be unique key and most critical info. Then use classifier for the other entries.